### PR TITLE
Kvssink pcm support

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kinesis-video-gstreamer-plugin/src/gstkvssink.cpp [hunk 1]
```diff
@@ -1274,7 +1290,27 @@ init_track_data(GstKvsSink *kvssink) {
...
+                g_free(kvssink->audio_codec_id);
+                kvssink->audio_codec_id = g_strdup(DEFAULT_AUDIO_CODEC_ID_PCM);
+                audio_content_type = g_strdup(DEFAULT_CONTENT_TYPE_AUDIO_ALAW);
+            } else if (strncmp(media_type, GSTREAMER_MEDIA_TYPE_MULAW, MAX_GSTREAMER_MEDIA_TYPE_LEN) == 0) {
+                g_free(kvssink->audio_codec_id);
+                kvssink->audio_codec_id = g_strdup(DEFAULT_AUDIO_CODEC_ID_PCM);
+                audio_content_type = g_strdup(DEFAULT_CONTENT_TYPE_AUDIO_MULAW);
+            } else {
+                // no-op, should result in a caps negotiation error before getting here.
+                LOG_AND_THROW("Code should not reach here");
```
Comment (MushMal) [inline kinesis-video-gstreamer-plugin/src/gstkvssink.cpp:1311]: LOL: the famous last words. At any rate, change this to an "Internal error with blah blah"

**Comment by wlinInspire** [comment]
@chehefen @MushMal 

My rtsp source has an audio track (PCM ALAW) and I am trying to send to kvssink using 
`rtsp. ! queue ! rtppcmadepay ! sink.` 
as part of the gstreamer pipeline but failed as shown below
```
2020-10-01 12:17:54 [140405422286592] WARN - putKinesisVideoFrame(): Failed to submit frame to Kinesis Video client. status: 0x32000002 decoding timestamp: 40970984 presentation timestamp: 16015798742564765
2020-10-01 12:17:54 [140405430679296] WARN - putKinesisVideoFrame(): Failed to submit frame to Kinesis Video client. status: 0x32000002 decoding timestamp: 41635695 presentation timestamp: 16015798743220427
2020-10-01 12:17:54 [140405422286592] WARN - putKinesisVideoFrame(): Failed to submit frame to Kinesis Video client. status: 0x30000005 decoding timestamp: 42322494 presentation timestamp: 16015798743892814
```

However, it works if I change transcode it to AAC using 
`rtsp. ! queue ! rtppcmadepay ! alawdec ! audioconvert ! avenc_aac ! sink.`

Track information is shown below from VLC player
image

**Comment by MushMal** [comment]
The first error 0x32000002 you are seeing is STATUS_MKV_INVALID_FRAME_TIMESTAMP which means that your stream is producing large timecodes that are not possible to encode in the MKV. Please make sure you have proper fragmentation setup - change your rtsp source to produce a key-frame between 1-6 seconds - the limit is 10 seconds.

The second issue 0x30000005 is caused by the first - hard to tell without debug logs.

**Comment by wlinInspire** [comment]
@MushMal I don't think it is related to the video as It is working when it is in video only mode.

**Comment by MushMal** [comment]
As I don't have any exposure to your custom device with your own custom media pipeline, I am just stating the facts about the warnings you are receiving. You might need to debug into your media pipeline. Try to get more verbose logs that might help you understand what's going on. Please cut a separate issue as this is a PR and won't be monitored. Include some details that can be actionable - as is, we won't be able to debug your device

**Comment by wlinInspire** [comment]
@MushMal Thanks, will do

**Pull request merged**